students/ericstreit/lessons/lesson03/assignment/create_customers.py

```diff
@@ -0,0 +1,54 @@
+"""
+#Lesson03
+#Customer Database Exercise ##
+"""
+
+#!/usr/bin/env python3
+#import stuff here
+from db_model import *
+import logging
+logging.basicConfig(level=logging.DEBUG)
+#global variables here
+
+#define function
+def create_customers(first__name, last__name, home__address, phone__number, email__address,
+                     account__status, credit__limit):
+    """
+    Description of function
+
+    :param arg1: The first very important parameter. And a bit about
+                 what it means.
+    :param arg2: The second very important parameter. And now some
+                 description of how this is used
+    etc
+    """
+    try:
+        with db.transaction():
+            logging.info(f'opening database: {db}')
+            customer = Customer.create(
+                    first_name = first__name,
+                    last_name = last__name,
+                    home_address = home__address,
+                    phone_number = phone__number,
+                    email_address = email__address,
+                    status = account__status,
+                    credit_limit = credit__limit)
+            customer.save()
+            logging.info('Database add successful')
+
+    except Exception as e:
+        logging.info(f'Hm, we had an error here?')
+        logging.info(e)
+    finally:
+        db.close()
```

Comment:
It would make more sense to put the try block just around the piece you think might cause an error, so Customer.create and customer.save, probably, rather than including everything. And then you let the user know what happened. I can see you already realize you may have errors you have not anticipated. ;-) So, if there is a problem with connecting to the database, rather than a problem with saving a customer, you probably want you customer to know that the database can't be reached at the moment, rather than just swallowing that error. If the problem is creating a customer, than you want them to know that there was something wrong with the information for that customer, but atm, in both cases, you are swallowing the error, and the user would not have any idea anything even went wrong.